pkg/apply/basic_printer.go

```diff
@@ -57,7 +69,46 @@ func (b *BasicPrinter) Print(ch <-chan Event) {
 	}
 }
 
+// PrintPreviewEvents outputs only preview events from the provided channel in a preview
+// format on StdOut.
+func (b *BasicPrinter) PrintPreviewEvents(ch <-chan Event) {
+	createdCnt := 0
+	modifiedCnt := 0
+	deletedCnt := 0
+	fmt.Fprintf(b.IOStreams.Out, "\nA preview of operations is shown below. Please use apply to perform the operations.\n\n")
+	for event := range ch {
+		obj := event.ApplyEvent.Object
+		gvk := obj.GetObjectKind().GroupVersionKind()
+		name := "<unknown>"
+		if acc, err := meta.Accessor(obj); err == nil {
+			if n := acc.GetName(); len(n) > 0 {
+				name = n
+			}
+		}
```

Comment:
please hide event parsing by factoring to tiny function  

`gvk, name := determineGvkN(event)`